### ISSUE TITLE
Installation issue: jdk can't download directly

### ISSUE THREAD

**Issue by sethrj** [LLNL/spack#26938]
### Steps to reproduce the issue

```console
$ spack install jdk
==> Installing jdk-14_36-a2wws5elorqs3dhhuy4uplhmkzfahcv6
==> No binary for jdk-14_36-a2wws5elorqs3dhhuy4uplhmkzfahcv6 found: installing from source
/rnsdhpc/code/spack/lib/spack/spack/target.py:137: UserWarning: microarchitecture specific optimizations are not supported yet on mixed compiler toolchains [check apple-clang@13.0.0 for further details]
  warnings.warn(msg.format(compiler))
==> Fetching https://download.oracle.com/otn-pub/java/jdk/14+36/076bab302c7b4508975440c56f6cc26a/jdk-14_linux-x64_bin.tar.gz
==> Warning: The contents of /var/folders/fy/x2xtwh1n7fn0_0q2kk29xkv9vvmbqb/T/s3j/spack-stage/spack-stage-jdk-14_36-a2wws5elorqs3dhhuy4uplhmkzfahcv6/jdk-14_linux-x64_bin.tar.gz look like HTML.  Either the URL you are trying to use does not exist or you have an internet gateway issue.  You can remove the bad archive using 'spack clean <package>', then try again using the correct URL.
==> Error: ChecksumError: sha256 checksum failed for /var/folders/fy/x2xtwh1n7fn0_0q2kk29xkv9vvmbqb/T/s3j/spack-stage/spack-stage-jdk-14_36-a2wws5elorqs3dhhuy4uplhmkzfahcv6/jdk-14_linux-x64_bin.tar.gz
    Expected 4639bbaecc9cc606f1a4b99fda1efcaefcbf57a7025b3828b095093a6c866afd but got 80212f7748c2671b89b6085000717c747851db004409dce3bb97f3a2aeb91cdd

/rnsdhpc/code/spack/lib/spack/spack/package.py:1392, in do_fetch:
       1389        self.stage.fetch(mirror_only, err_msg=err_msg)
       1390        self._fetch_time = time.time() - start_time
       1391
  >>   1392        if checksum and self.version in self.versions:
       1393            self.stage.check()
       1394
       1395        self.stage.cache_local()
```

The gzip file is HTML that reads:
```
Sorry!
 
In order to download products from Oracle Technology Network you must agree to the OTN license terms.
  
Be sure that...
- Your browser has "cookies" and JavaScript enabled.
- You clicked on "Accept License" for the product you wish to download.
- You attempt the download within 30 minutes of accepting the license.
```

### Information on your system

* **Spack:** 0.16.3-4967-3720d06e26
* **Python:** 3.8.8
* **Platform:** darwin-bigsur-skylake
* **Concretizer:** clingo

### Additional information

not applicable

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

**Comment by Pandapip1** [comment]
I have the same issue. I solved it by adding a download to openjdk.

**Comment by bernhardkaindl** [comment]
fixed by #43264
